### PR TITLE
Add Speedtest presence

### PR DESCRIPTION
**Pull request by Nevylish**
Hello, I made a presence for https://www.speedtest.net/

Screenshot_2
Screenshot_3
Screenshot_4
Screenshot_5
Screenshot_6
Screenshot_7
Screenshot_9

### CODE REVIEW

File: websites/S/Speedtest/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "description": {
+    "de": "Der umfassende Breitband-Geschwindigkeitstest",
+    "en": "The Global Broadband Speed Test",
+    "es": "La prueba de velocidad de banda ancha global",
+    "fr": "Le test de vitesse de connexion global"
+  },
+  "service": "Speedtest",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/MOxmNLG.png",
+  "thumbnail": "https://i.imgur.com/MdYNATM.png",
```
Comment (ririxidev) [inline websites/S/Speedtest/dist/metadata.json:17]: Pre``MID`` telecom yes.

File: websites/S/Speedtest/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  ],
+  "category": "other",
+  "settings": [
+    {
+      "id": "showISP",
+      "title": "Show ISP Name",
+      "icon": "fad fa-user-secret",
+      "value": true
+    }
+  ]
```
Comment (Bas950) [inline websites/S/Speedtest/dist/metadata.json:32]: probs better to set default to false

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/S/Speedtest/dist/metadata.json (hunk 1 above)
Comment (ririxidev) [reply websites/S/Speedtest/dist/metadata.json:17]: technically, fix it lol

**Review by Bas950 — APPROVED** [review]

File: websites/S/Speedtest/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  "description": {
+    "de": "Der umfassende Breitband-Geschwindigkeitstest",
+    "en": "The Global Broadband Speed Test",
+    "es": "La prueba de velocidad de banda ancha global",
+    "fr": "Le test de vitesse de connexion global"
+  },
+  "service": "Speedtest",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/MOxmNLG.png",
+  "thumbnail": "https://i.imgur.com/JXIDlOL.png",
```
Comment (StrikerFRFX) [inline websites/S/Speedtest/dist/metadata.json:17]: While it's nice that the thumbnail is PreMiD Telecom and PreMiD.app as the URL, I'd just suggest changing it to different values (i.e a massive Telecom Provider), and then for the I.P just put 192.168.0.1 or 127.0.0.1 or localhost, just something not your IP obviously for privacy concerns.

**Review by StrikerFRFX — COMMENTED** [review]
While it's nice that the thumbnail is PreMiD Telecom and PreMiD.app as the URL, I'd just suggest changing it to different values (i.e a massive Telecom Provider), and then for the I.P just put 192.168.0.1 or 127.0.0.1 or localhost, just something not your IP obviously for privacy concerns.

File: websites/S/Speedtest/presence.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      ping = document.querySelector(".result-item-ping .result-data").textContent + document.querySelector(".result-item-ping .result-data-unit").textContent,
+      download = document.querySelector(".result-item-download .result-data").textContent + document.querySelector(".result-item-download .result-data-unit").textContent,
+      upload = document.querySelector(".result-item-upload .result-data").textContent + document.querySelector(".result-item-upload .result-data-unit").textContent;
+      let isp = document.querySelector(".result-item-icon.result-item-isp .result-label").textContent;
+      if (!showISP) isp = "Hidden";
+      // 27 & 29 = nothing + unit
+      if (upload.length == 29) {
+        presenceData.details = `Live results - Ping ${ping.length == 27 ? "Testing" : ping} | Download ${download.length == 29 ? "Testing" : download} | Upload ${upload.length == 29 ? "Testing" : upload}`;
+        presenceData.state = `ISP: ${isp} | Server: ${server}`;
```
Comment (StrikerFRFX) [inline websites/S/Speedtest/presence.ts:22]: could use a ternary operator for this to possibly make it show only `Server: {}` when their ISP is hidden.

**Review by StrikerFRFX — APPROVED** [review]

**Pull request merged**